doc/CMakeLists.txt

```diff
@@ -28,14 +28,29 @@ if(BUILD_DOCUMENTATION)
 
     set(srcdir ${CMAKE_CURRENT_SOURCE_DIR})
     set(VERSION ${GEOS_VERSION_FULL})
-    set(DOXYGEN_LOGFILE ${CMAKE_CURRENT_BINARY_DIR}/doxygen_log)
-    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile.in
-                   ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile
-                   @ONLY)
+    set(DOXYGEN_LOGFILE ${CMAKE_CURRENT_BINARY_DIR}/doxygen.log)
+    set(CHECK_ERROR_SCRIPT "check_doxygen_errors.cmake")
+    configure_file(
+        "${CMAKE_CURRENT_SOURCE_DIR}/${CHECK_ERROR_SCRIPT}"
+        "${CMAKE_CURRENT_BINARY_DIR}/${CHECK_ERROR_SCRIPT}"
+        COPYONLY)
+    configure_file(
+        ${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile.in
+        ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile
+        @ONLY)
     unset(srcdir)
     unset(VERSION)
-    unset(DOXYGEN_LOGFILE)
 
     add_custom_target(docs
-            COMMAND ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile)
+        COMMAND ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile
```

Comment:
What do you think about registering this as a test, like this?

```
    add_custom_target(docs
        COMMAND ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile
        DEPENDS geos
        BYPRODUCTS ${DOXYGEN_LOGFILE})

    add_test(NAME test_docs
             COMMAND ${CMAKE_COMMAND}
                 -D DOXYGEN_LOGFILE=${DOXYGEN_LOGFILE}
                 -P "${CMAKE_CURRENT_BINARY_DIR}/${CHECK_ERROR_SCRIPT}")
```

Reply:
Makes sense, a test is after all what it is. Updated as recommended.
Thanks!